src/components/Netcat/Netcat.tsx

```diff
@@ -48,16 +51,94 @@ const NetcatTool = () => {
+            // If the process was successful, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+
+                // If the process was terminated manually, display a termination message.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+
+                // If the process was terminated with an error, display the exit and signal codes.
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
     const onSubmit = async (values: FormValuesType) => {
         //Starts off with the IP address after netcat
         //Ex: nc <ip address>
         let args = [``];
 
         //If verbose mode is checked, v flag is added to args
+        //These two lines are used for different command structures so typos don't occur in the presence or absence of verbose mode
+        //A task for future students could be to reduce it to one line of code that works across all commands without creating syntax errors
         const verboseFlag = checkedVerboseMode ? "v" : "";
+        const verboseFlagWithSpaceAndDash = checkedVerboseMode ? " -v" : "";
 
         //Switch case
         switch (values.netcatOptions) {
+            case "Listen": //Sets up nc listener, nc syntax: nc -lvp <port number>
+                args = [`-l${verboseFlag}p`];
+                args.push(values.portNumber);
+
+                CommandHelper.runCommandGetPidAndOutput("nc", args, handleProcessData, handleProcessTermination)
+                    .then(({ output, pid }) => {
+                        // Update the UI with the results from the executed command
+                        setOutput(output);
+                        console.log(pid);
+                        setPid(pid);
+                    })
+                    .catch((error) => {
+                        // Display any errors encountered during command execution
+                        setOutput(error.message);
+                        // Deactivate loading state
+                        setLoading(false);
+                    });
+
+                break;
+
+            case "Connect": //Connects to an nc listener, nc syntax: nc -v <ip address> <port number>
+                let command = `nc${verboseFlagWithSpaceAndDash} ${values.ipAddress} ${values.portNumber}`;
+
+                //CommandHelper.runCommandGetPidAndOutput("nc", args, handleProcessData, handleProcessTermination)
+                CommandHelper.runCommandGetPidAndOutput(
+                    "bash",
```

Comment:
I am not sure why you need to use bash here. The command nc <IP address> <port> should work without bash. It's probably worth a comment to explain. 

- [ ] Comment required to explain the use of bash.